torch/csrc/jit/codegen/cuda/glfdc/stack.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include <limits>
+#include <stack>
+#include <tuple>
+#include <utility>
+#include <vector>
+
+namespace torch {
+namespace jit {
+namespace fuser {
+namespace cuda {
+namespace glfdc {
+
+// Index and cookie of value to replace
+using BindingGapType = std::pair<std::size_t, uintptr_t>;
+using BindCookieType = typename std::tuple_element<1, BindingGapType>::type;
+
+// EvalStack - Simple std::stack extension for use in stack machine evaluation.
+template <typename Ty_, typename C_ = std::vector<Ty_>>
+class EvalStack : std::stack<Ty_, C_> {
+  using base_t = std::stack<Ty_, C_>;
+
+ public:
+  static constexpr Ty_ GAP_VALUE = std::numeric_limits<Ty_>::max();
+
+  EvalStack() = default;
+
+  EvalStack(const EvalStack&) = default;
+  EvalStack(EvalStack&&) = default;
+
+  EvalStack& operator=(const EvalStack&) = default;
+  EvalStack& operator=(EvalStack&&) = default;
+
+  using base_t::emplace;
+  using base_t::pop;
+  using base_t::push;
+
+  // pop_top - Shorthand for stack.top(); stack.pop();
+  Ty_ pop_top() {
+    TORCH_INTERNAL_ASSERT(!this->c.empty());
+    Ty_ ret = this->c.back();
+    pop();
+    return ret;
+  }
+
+  using base_t::swap;
+
+  // drop - drops n operands from stack
+  void drop(std::size_t n) noexcept {
+    TORCH_CHECK(n <= this->c.size());
+    this->c.resize(this->c.size() - n);
```

Comment:
I'm not sure how confident we are with the stability of this protected variable of `std::stack` over time. Is there any doc from `libstdc++` or `libc++` about this `c` existing long term? The implementation make sense but just the protected variable naming `c` from `std::stack` did not sound particularly user-facing. cc @naoyam @csarofeen 

Also since `std::stack` is supposed to be a wrapper class, would it make sense to include our "local stable" version in the repo instead of inheriting the `std` one?

Reply:
Since this protected member it's been part of HP STL released in 1994 and been part of standard to this day (c++20) I doubt we should be concerned about it. Please see N4868 `22.6.6.2 [stack.defn]`. True, it might be deprecated in c++23 and then removed in c++2z but that can be said about anything else.